pkg/compiler/transformers/k8s/workflow.go

```diff
@@ -207,14 +207,20 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 		NodeDefaults: v1alpha1.NodeDefaults{Interruptible: interruptible},
 	}
 
-	obj.ObjectMeta.Name, obj.ObjectMeta.GenerateName, obj.ObjectMeta.Labels[ExecutionIDLabel], err =
-		generateName(wf.GetId(), executionID)
+	obj.ObjectMeta.Name, obj.ObjectMeta.GenerateName, obj.ObjectMeta.Labels[ExecutionIDLabel], obj.ObjectMeta.Labels[ProjectLabel],
+		obj.ObjectMeta.Labels[DomainLabel], err = generateName(wf.GetId(), executionID)
 
 	if err != nil {
 		errs.Collect(errors.NewWorkflowBuildError(err))
 	}
 	obj.ObjectMeta.Labels[WorkflowNameLabel] = utils.SanitizeLabelValue(WorkflowNameFromID(primarySpec.ID))
 
+	h := fnv.New32a()
+	h.Write([]byte(obj.ObjectMeta.Labels[ExecutionIDLabel]))
+	hash := h.Sum32() % 32
```

Comment:
can we make 32 to be a global const if not configurable?

Reply:
done. it was in the original sharding pr, i was unsure where to put it in this one - it will certainly move.